### PR TITLE
Netlink route support

### PR DESCRIPTION
**Pull request by Meissi-jian**
## Summary
Support route/neigh/prefix netlink notify

## Impact
Will send netlink msg when route/neigh/prefix changs

## Testing
Passed the test under simulated environment

### CODE REVIEW

**Review by acassis — COMMENTED** [review]
@Meissi-jian please include Documentation/ to this Netlink route support!

This way someone looking our documentation page will discover it is supported.

File: net/netlink/netlink_route.c [hunk 1]
```diff
@@ -708,23 +707,46 @@ static int netlink_ipv4_route(FAR struct net_route_ipv4_s *route,
...
 
+  return (FAR struct netlink_response_s *)alloc;
+}
+
+/****************************************************************************
+ * Name: netlink_ipv4route_callback
+ *
+ * Description:
+ *   Response netlink message from ipv4 route list.
+ *
```
Comment (acassis) [inline net/netlink/netlink_route.c:809]: Please include the Input Parameters

File: net/netlink/netlink_route.c [hunk 2]
```diff
@@ -801,9 +822,33 @@ static int netlink_ipv6_route(FAR struct net_route_ipv6_s *route,
...
 
+  return (FAR struct netlink_response_s *)alloc;
+}
+
+/****************************************************************************
+ * Name: netlink_ipv6route_callback
+ *
+ * Description:
+ *   Response netlink message from ipv6 route list.
+ *
```
Comment (acassis) [inline net/netlink/netlink_route.c:924]: Please include Input Parameters

**Review by acassis — CHANGES_REQUESTED** [review]
Please include Documentation/

**Comment by acassis** [comment]
@wengzhe could you please @Meissi-jian to write the Documentation/ ?

**Review by acassis — APPROVED** [review]

**Comment by acassis** [comment]
Thank you very much @Meissi-jian !

Question: will it be compatible with Linux netlink, i.e.: https://olegkutkov.me/2019/08/29/modifying-linux-network-routes-using-netlink/

**Comment by acassis** [comment]
@Meissi-jian please fix this error:

```
In function 'netlink_get_neighbor',
    inlined from 'netlink_get_neighborlist' at netlink/netlink_route.c:736:10,
    inlined from 'netlink_route_sendto' at netlink/netlink_route.c:1380:19:
Error: netlink/netlink_route.c:703:7: error: argument 2 null where non-null expected [-Werror=nonnull]
  703 |       memcpy(resp->data, neigh, tabsize);
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from netlink/netlink_route.c:28:
/github/workspace/sources/nuttx/include/string.h: In function 'netlink_route_sendto':
/github/workspace/sources/nuttx/include/string.h:101:12: note: in a call to function 'memcpy' declared 'nonnull'
  101 | FAR void  *memcpy(FAR void *dest, FAR const void *src, size_t n);
      |            ^~~~~~
cc1: all warnings being treated as errors
make[1]: *** [Makefile:79: netlink_route.o] Error 1
make[1]: Target 'libnet.a' not remade because of errors.
make: *** [tools/LibTargets.mk:71: net/libnet.a] Error 2
make: Target 'all' not remade because of errors.
```

**Comment by Meissi-jian** [comment]
> Thank you very much @Meissi-jian !
> 
> Question: will it be compatible with Linux netlink, i.e.: https://olegkutkov.me/2019/08/29/modifying-linux-network-routes-using-netlink/

hi @acassis the current status is as follows, i think the difference is the data content of our route/arp/neigh tables. 
1. RTMGRP_IPVx_ROUTE : the header and rtmsg is same,the rtattr data is incompatible.And this pull just add notify.
2. RTMGRP_NEIGH: same as above.
3. RTNLGRP_IPV6_PREFIX: new added, compatible with linux

**Review by jerpelea — APPROVED** [review]
please add a commit message

**Comment by Meissi-jian** [comment]
> please add a commit message

@jerpelea sorry,i catn't get it, add a message about what?

**Comment by acassis** [comment]
I think all PRs title have enough info and he added Documentation with more details.

**Pull request merged**